.github/workflows/ci.yml

```diff
@@ -0,0 +1,86 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches: [ master, dev ]
+  schedule: [ cron: "0 6 * * 4" ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  test:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install stable Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+
+      - name: Get Cargo version
+        id: cargo_version
+        run: echo "::set-output name=version::$(cargo -V | tr -d ' ')"
+        shell: bash
+
+      - name: Download cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/
```

Comment:
Won't the rust install have set this up? Why cache as well?

Reply:
I was following caching suggestions from The Cargo Book:
https://doc.rust-lang.org/cargo/guide/cargo-home.html#caching-the-cargo-home-in-ci

Which directories do you think are redundant?
We can change those in this PR and open an issue with The Cargo Book to unlist those.

Reply:
`.cargo/bin/` I will asked on https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/why.20cache.20.2Ecargo.2Fbin.2F.20.3F

It it is recommended by the book then lets do it. If the book is wrong for our needs we can do a follow up PR.

Reply:
We can wait for the decision that comes out of that thread.
In our case I don't think caching that makes a difference since we don't use `cargo install` and the dir is going to be empty, so might as well remove it from caching.